dbt_semantic_interfaces/dataclass_serialization.py

```diff
@@ -221,11 +225,14 @@ def _convert_dataclass_instance_to_pydantic_model(
         return obj
 
     def pydantic_serialize(self, obj: SerializableDataclassT) -> str:  # noqa: D
-        assert dataclasses.is_dataclass(obj)
+        obj_class = obj.__class__
+        assert dataclasses.is_dataclass(obj), f"Got object of type: {obj.__class__.__name__}"
+        assert isinstance(obj, SerializableDataclass), f"Got object of type: {obj.__class__.__name__}"
+        assert issubclass(obj_class, SerializableDataclass), f"Got object type: {obj_class.__name__}"
 
         return self._convert_dataclass_instance_to_pydantic_model(
             # .__class__ seems to be the approach for new classes and there are differences with type(obj)
```

Comment:
Move this comment to where the local is defined?

Reply:
Updated.